### PR TITLE
Feature/MN-273/native-demo-app

### PR DESCRIPTION
**Pull request by rohitsharma120582**
## Technical description
- [x] Create a new package named `@monkvision/eslint-config-typescript-react-native` for ESLint config that will be the one used everywhere in React Native project in MonkJs
- [x] Create a blank NATIVE demo app to test the refacto components
- [x] Run and test app in Android emulator

### CODE REVIEW

**Comment by rohitsharma120582** [comment]
Fixed all ESLint issues from the code...

**Review by deepakglobant — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -17,6 +17,10 @@
 node_modules
 .idea
 
+# production
+apps/monk-test-app/build
+apps/monk-test-app-native/build
+
```
Comment (souyahia-monk) [inline .gitignore:23]: Remove these lines from the gitignore at the root of the repo and create 2 gitignores inside the apps/monk-test-app and apps/monk-test-app-native directories where you add the build param

File: apps/monk-test-app-native/.eslintignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/node_modules
```
Comment (souyahia-monk) [inline apps/monk-test-app-native/.eslintignore:1]: Add a line at the end of the file

File: apps/monk-test-app-native/.prettierrc.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  arrowParens: 'avoid',
+  bracketSameLine: true,
+  bracketSpacing: false,
+  singleQuote: true,
+  trailingComma: 'all',
+};
```
Comment (souyahia-monk) [inline apps/monk-test-app-native/.prettierrc.js:7]: Use Monk prettier config instead

File: apps/monk-test-app-native/.watchmanconfig [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (souyahia-monk) [inline apps/monk-test-app-native/.watchmanconfig:1]: Add a new line at the end of this file

File: apps/monk-test-app/package.json [hunk 5]
```diff
@@ -53,11 +59,7 @@
     "eslint-plugin-promise": "^6.1.1",
     "eslint-plugin-react": "^7.32.2",
     "eslint-plugin-react-hooks": "^4.6.0",
-    "eslint-utils": "^3.0.0",
-    "jest": "^29.3.1",
-    "prettier": "^2.7.1",
-    "regexpp": "^3.2.0",
-    "ts-jest": "^29.0.3"
+    "eslint-utils": "^3.0.0"
```
Comment (souyahia-monk) [inline apps/monk-test-app/package.json:62]: Why are all these dependencies declared in the "peerDependencies" by the way? I think they should be decalred in the regular dependencies

File: configs/typescript-config/tsconfig.reactnative.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (souyahia-monk) [inline configs/typescript-config/tsconfig.reactnative.json:1]: Rename this file to `tsconfig.react-native.json`

File: .gitignore (hunk 1 above)
Comment (rohitsharma120582) [reply .gitignore]: Done

File: apps/monk-test-app-native/.eslintignore (hunk 2 above)
Comment (rohitsharma120582) [reply apps/monk-test-app-native/.eslintignore]: Done

File: apps/monk-test-app-native/.prettierrc.js (hunk 3 above)
Comment (rohitsharma120582) [reply apps/monk-test-app-native/.prettierrc.js]: Done

File: apps/monk-test-app-native/.watchmanconfig (hunk 4 above)
Comment (rohitsharma120582) [reply apps/monk-test-app-native/.watchmanconfig]: Done

File: apps/monk-test-app-native/app.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "MonkTestAppNative",
+  "displayName": "MonkTestAppNative"
+}
```
[the remark this answers is not in the document]
Comment (rohitsharma120582) [reply apps/monk-test-app-native/app.json]: Done

File: configs/typescript-config/tsconfig.reactnative.json (hunk 6 above)
Comment (rohitsharma120582) [reply configs/typescript-config/tsconfig.reactnative.json]: Done

File: apps/monk-test-app/package.json (hunk 5 above)
Comment (rohitsharma120582) [reply apps/monk-test-app/package.json:62]: Yes, we should defined these dependencies as **"peerDependencies"** in test application. As project throws following errors, if place them into regular dependencies:
* Application build gets failed for React-Native demo test app as **"@react-native-community"** internally uses ESLint packages and demand them from project's **"node_modules"** folder itself
* ESLint error when using ESLint and TypeScript dependencies as regular dependencies. Sharing one error message for the reference:
```
ESLint couldn't determine the plugin "jsx-a11y" uniquely

/Users/rohit.sharma/workspace/projects/ACV/monkjs_refacto/node_modules/eslint-plugin-jsx-a11y/lib/index.js (loaded in ".eslintrc.js » @monkvision/eslint-config-typescript-react-native")

/Users/rohit.sharma/workspace/projects/ACV/monkjs_refacto/apps/monk-test-app-native/node_modules/eslint-plugin-jsx-a11y/lib/index.js (loaded in ".eslintrc.js » @monkvision/eslint-config-typescript-react-native » plugin:jsx-a11y/recommended")
```

Comment (rohitsharma120582) [reply apps/monk-test-app/package.json:62]: This type of fix might require sometime for research. So we can take care of this rectification later.

**Review by souyahia-monk — APPROVED** [review]

**Pull request merged**